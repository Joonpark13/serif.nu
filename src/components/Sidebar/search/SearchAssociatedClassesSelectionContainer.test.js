import { fromJS } from 'immutable';
import {
  currentCourseNameSelector,
  currentSectionNumberSelector,
  currentAssociatedClassesSelector,
} from './SearchAssociatedClassesSelectionContainer';

describe('currentCourseNameSelector', () => {
  it('should select current course name from state', () => {
    const currentCourseName = 'Some Course Name';
    const testState = fromJS({
      search: {
        currentCourseName,
      },
    });

    expect(currentCourseNameSelector(testState)).toEqual(currentCourseName);
  });
});

describe('currentSectionNumberSelector', () => {
  it('should select current section number from state', () => {
    const currentSectionNumber = '78719';
    const testState = fromJS({
      search: {
        currentSectionNumber,
      },
    });

    expect(currentSectionNumberSelector(testState)).toEqual(currentSectionNumber);
  });
});

describe('currentAssociatedClassesSelector', () => {
  it('should select current section number from state', () => {
    const currentAssociatedClasses = [];
    const testState = fromJS({
      search: {
        currentAssociatedClasses,
      },
    });

    expect(currentAssociatedClassesSelector(testState)).toEqual(fromJS(currentAssociatedClasses));
  });
});
