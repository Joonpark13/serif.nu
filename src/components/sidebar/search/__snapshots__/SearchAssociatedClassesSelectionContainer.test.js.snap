// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchAssociatedClassesSelectionContainer renders 1`] = `
<AssociatedClassesSelection
  addSectionWithAssociatedClass={[Function]}
  associatedClassHover={[Function]}
  associatedClassHoverOff={[Function]}
  associatedClasses={
    Array [
      Object {},
    ]
  }
  back={[Function]}
  currentCourseName="course name"
  currentSectionNumber="section num"
/>
`;
